internal/mesh/internal/controllers/gatewayproxy/controller.go

```diff
@@ -104,7 +112,25 @@ func (r *reconciler) Reconcile(ctx context.Context, rt controller.Runtime, req c
 		rt.Logger.Trace("proxy state template for this gateway doesn't yet exist; generating a new one")
 	}
 
-	newPST := builder.NewProxyStateTemplateBuilder(workload).Build()
+	exportedServicesID := &pbresource.ID{
+		Name:    "global",
+		Tenancy: &pbresource.Tenancy{},
```

Comment:
I believe this should specify the partition that we're operating in
```suggestion
		Tenancy: &pbresource.Tenancy{
			Partition: req.ID.Tenancy.Partition,
		},
```